File: 128-LongestConsecutiveSequence/128-LongestConsecutiveSequence.java
// Last updated: 2/27/2026, 5:22:15 PM
class Solution {
    public int longestConsecutive(int[] nums) {

        if(nums.length == 0) return 0;

        HashSet<Integer> set = new HashSet<>();

        for(int i : nums) set.add(i);

        int max = Integer.MIN_VALUE;

        for(int i = 0; i < nums.length; i++){

            if(!set.contains(nums[i] - 1)){
                int curr = nums[i];
                int currLength = 0;
                while(set.contains(curr)) {
                     curr++;
                     currLength++;
                }

                max = Math.max(currLength, max);
            }


        }

        return max;

    }
}

File: 128-LongestConsecutiveSequence/128-LongestConsecutiveSequence.py
# Last updated: 2/28/2026, 4:28:32 PM
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:

        if not nums: return 0

        longest = 1
        st = set(nums)

        for num in st:
            if (num - 1) in st:
                continue
            else:
                inc = 1
                curr = 1
                while num + inc in st:
                    curr+=1
                    longest = max(curr, longest)
                    inc+=1
                


        return longest
        